src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -199,17 +201,17 @@ namespace xrt {
     ffi_arg kernel_return = 0;
     std::vector<void*> ffi_arg_values(m_kernel_args.size());
     std::vector<ps_arg> bo_args;
-    clockc::time_point start;
-    clockc::time_point end;
-    clockc::time_point cmd_start;
-    clockc::time_point cmd_end;
+    clockc::time_point start = clockc::now();
+    clockc::time_point end = clockc::now();
+    clockc::time_point cmd_start = clockc::now();
+    clockc::time_point cmd_end = clockc::now();
```

Comment:
Why 4 calls to the operating system? Are you doing some statistics on the system clock? :-)